cdap-ui/app/features/tracker/directives/histogram/histogram.js

```diff
@@ -18,115 +18,117 @@ angular.module(PKG.name + '.feature.tracker')
-        .attr('class', 'y axis')
-        .call(yAxis);
-
-
-      /* CREATING TOOLTIP */
-      let tip = d3.tip()
-        .attr('class', 'd3-tip tracker-audit-histogram')
-        .offset([-10, 0])
-        .html(function(d) {
-          return '<strong>' + timeFormat(d.time) + ' - ' + d.count +  '</strong>';
-        });
-      svg.call(tip);
-
-      /* BARS */
-      svg.selectAll('.bar')
-          .data(data)
-        .enter().append('rect')
-          .attr('class', 'bar')
-          .attr('x', (d) => { return x(d.time) - (width / data.length); })
-          .attr('y', (d) => { return y(d.count); })
-          .attr('width', width/data.length - (width * 0.2 / data.length ))
-          .attr('height', (d) => { return height - y(d.count); })
-        .on('mouseover', tip.show)
-        .on('mouseout', tip.hide);
-
+        let x = d3.time.scale()
+          .range([
+            (width * 0.2 / data.length / 2) + (width / data.length),
+            width - (width / data.length / 2)
+          ]);
+
+        let y = d3.scale.linear()
+          .range([height, 0]);
+
+        let parentContainer = d3.select(element[0]).select('.histogram-container')
+          .style({
+            height: parentHeight + 'px'
+          });
+
+        let svg = parentContainer.append('svg')
+            .attr('width', width + margin.left + margin.right)
+            .attr('height', parentHeight)
+          .append('g')
+            .attr('transform', 'translate(' + margin.left + ', ' + margin.top + ')');
+
+        /* CREATE GRAPH */
+        x.domain([data[0].time, data[data.length - 1].time]);
+        y.domain([0, d3.max(data, (d) => { return d.count; })]).nice();
+
+        /* X AXIS */
+        let timeFormat = d3.time.format('%b %d, %Y');
+        let xAxis = d3.svg.axis()
+          .scale(x)
+          .orient('bottom')
+          .tickPadding(10)
+          .tickFormat(timeFormat)
+          .outerTickSize(0)
+          .innerTickSize(0)
+          .ticks(d3.time.day, 1);
+
+        svg.append('g')
+          .attr('class', 'x axis')
+          .attr('transform', 'translate(0, ' + height + ')')
+          .call(xAxis);
+
+        /* Y AXIS */
+        let yAxis = d3.svg.axis()
+          .scale(y)
+          .orient('left')
+          .tickPadding(10)
+          .outerTickSize(0)
+          .innerTickSize(-width)
+          .ticks(5);
+
+        svg.append('g')
+          .attr('class', 'y axis')
+          .call(yAxis);
+
+        /* CREATING TOOLTIP */
+        let tip = d3.tip()
+          .attr('class', 'd3-tip tracker-audit-histogram')
+          .offset([-10, 0])
+          .html(function(d) {
+            return '<strong>' + timeFormat(d.time) + ' - ' + d.count +  '</strong>';
+          });
+        svg.call(tip);
+
+        /* BARS */
+        svg.selectAll('.bar')
+            .data(data)
+          .enter().append('rect')
+            .attr('class', 'bar')
+            .attr('x', (d) => { return x(d.time) - (width / data.length); })
```

Comment:
If its a single statement function that just returns something then we could have a concise representation like this,
```
.attr('x', d => x(d.time) - (width / data.length))
```